test/setup-test-env.ts

```diff
@@ -1,15 +1,26 @@
 import { installGlobals } from "@remix-run/node";
 import "@testing-library/jest-dom/extend-expect";
-import { expect, afterEach } from "vitest";
+import { afterEach, beforeAll, expect } from "vitest";
 import { cleanup } from "@testing-library/react";
 import matchers from "@testing-library/jest-dom/matchers";
+import { server } from "../mocks/server";
 
 installGlobals();
 
 // extends Vitest's expect method with methods from react-testing-library
 expect.extend(matchers);
 
+const closeServer = () => server.close();
+
+beforeAll(() => {
+  server.listen({ onUnhandledRequest: "bypass" });
+  process.once("SIGINT", closeServer);
+  process.once("SIGTERM", closeServer);
+  return closeServer;
```

Comment:
I'm a little confused why we're closing the server immediately after starting it? This gets run right after this hook and before all tests from what I can tell.

Reply:
[the returned function of `beforeAll` is essentially the `afterAll`.](https://vitest.dev/api/#beforeall)

Reply:
Ah okay, gotcha. I did a manual test and it seemed like it called the return function before it ended. No worries though! I see what you're saying based on the documentation you shared.